src/coreclr/scripts/superpmi.py

```diff
@@ -473,6 +498,23 @@ def read_csv_metrics(path):
 
     return None
 
+def determine_clrjit_compiler_version(clrjit_path):
+    """ Obtain the version of the compiler that was used to compile the clrjit at the specified path.
+
+    Returns:
+        A string extract from the binary or "unknown" if the string could not be found.
+        See 'jitBuildString' in buildstring.cpp in the JIT to see where this is defined.
+    """
+
+    with open(clrjit_path, "rb") as fh:
+        contents = fh.read()
```

Comment:
So this isn't actually reading the string via the export added in https://github.com/dotnet/runtime/pull/68329; it's actually reading the entire JIT native code file and parsing for this string. That seems... indirect (slow, also?).

I wonder if it would be better to do something like what was done to determine the JIT-EE version the JIT was built with by adding `-printJITEEVersion` as an argument to the SPMI utility `mcs` (https://github.com/dotnet/runtime/pull/42238). The idea being, the JIT and mcs will be built in the same build by the same toolset, so we can use `mcs` as a proxy for how the JIT is built. Then, here you could do something like run `mcs -printBuildString`. This is already done in this file in `determine_jit_ee_version()`.

Reply:
Seems unlikely to me that this kind of search would be slower than actually creating a process, mapping the images, etc. It is a simple text search through 1.5 MB of text. And is invoking mcs.exe to have it load the library to get the export really any more direct?

I agree that ideally we would parse the library, but in this case I was just being pragmatic about it -- this is simple and works in a handful of lines. The main reason I made it an export was just to make sure it is not removed by the compiler (and also, it sort of makes sense to have it when it does get consumed as if it was an export).

Reply:
> The idea being, the JIT and mcs will be built in the same build by the same toolset, so we can use mcs as a proxy for how the JIT is built. 

I'm not really sure how it works with the JIT-EE GUID right now, but I don't see how we would get the compiler version of the baseline this way.

Reply:
The way you've done it is reasonable, and pragmatic. I suppose I could have done something similar with the JIT-EE GUID as a text string stashed in the JIT dll, as an exported string and/or an embedded string with enough unique context such that it's parseable. Having it as output from running `mcs` means I can look at it by easily running a command-line tool, which is nice.

The GUID is used to determine the MCH files to use; it's unrelated to finding a baseline.